File: CodigosC/Banco/ContaBancaria.c
#include <stdio.h>

struct ContaBancaria {
    char titular[50];
    float saldo;
};

void depositar(struct ContaBancaria *conta, float valor) {
    conta->saldo += valor;
    printf("Depósito de R$%.2f realizado com sucesso.\n", valor);
}

void retirar(struct ContaBancaria *conta, float valor) {
    if (valor > conta->saldo) {
        printf("Erro: Saldo insuficiente para realizar a retirada de R$%.2f.\n", valor);
    } else {
        conta->saldo -= valor;
        printf("Retirada de R$%.2f realizada com sucesso.\n", valor);
    }
}

void consultarSaldo(struct ContaBancaria conta) {
    printf("Saldo atual: R$%.2f\n", conta.saldo);
}

int main() {
    struct ContaBancaria conta;
    int opcao;
    float valor;

    // Solicitar o nome do titular e o saldo inicial
    printf("Digite o nome do titular da conta: ");
    fgets(conta.titular, 50, stdin);
    conta.saldo = 0;

    do {
        // Exibir menu de opções
        printf("\n--- MENU ---\n");
        printf("1. Depositar\n");
        printf("2. Retirar\n");
        printf("3. Consultar Saldo\n");
        printf("4. Sair\n");
        printf("Escolha uma opção: ");
        scanf("%d", &opcao);

        switch (opcao) {
            case 1:
                printf("Digite o valor a ser depositado: ");
                scanf("%f", &valor);
                depositar(&conta, valor);
                break;
            case 2:
                printf("Digite o valor a ser retirado: ");
                scanf("%f", &valor);
                retirar(&conta, valor);
                break;
            case 3:
                consultarSaldo(conta);
                break;
            case 4:
                printf("Saindo do sistema...\n");
                break;
            default:
                printf("Opção inválida. Tente novamente.\n");
        }
    } while (opcao != 4);

    return 0;
}

File: CodigosPython/Dineiro/ContaBancaria.py

class ContaBancaria:
    def __init__(self, titular, saldo_inicial=0):
        self.titular = titular
        self.saldo = saldo_inicial

    def depositar(self, valor):
        if valor > 0:
            self.saldo += valor
            print(f"Depósito de R${valor:.2f} realizado com sucesso!")
        else:
            print("O valor do depósito deve ser positivo.")

    def sacar(self, valor):
        if valor > self.saldo:
            print("Saldo insuficiente para o saque.")
        elif valor <= 0:
            print("O valor do saque deve ser positivo.")
        else:
            self.saldo -= valor
            print(f"Saque de R${valor:.2f} realizado com sucesso!")

    def exibir_saldo(self):
        print(f"Saldo atual de {self.titular}: R${self.saldo:.2f}")


print("Bem-vindo ao sistema bancário!")
titular = input("Digite o nome do titular da conta: ")
conta = ContaBancaria(titular)

while True:
    print("\nEscolha uma operação:")
    print("1. Exibir saldo")
    print("2. Depositar")
    print("3. Sacar")
    print("4. Sair")
    opcao = input("Digite o número da opção: ")

    if opcao == "1":
        conta.exibir_saldo()
    elif opcao == "2":
        valor = float(input("Digite o valor a ser depositado: "))
        conta.depositar(valor)
    elif opcao == "3":
        valor = float(input("Digite o valor a ser sacado: "))
        conta.sacar(valor)
    elif opcao == "4":
        print("Obrigado por usar o sistema bancário! Até mais!")
        break
    else:
        print("Opção inválida. Tente novamente.")
